HealthMate/HealthMate.DAL/Repositories/GenericRepository.cs

```diff
@@ -42,7 +49,8 @@ public async Task DeleteAsync(Guid id, CancellationToken token)
             }
```

Comment:
Are you sure that you want to use EF's tracking and turn off it on for some situations? You can disable in in DbContext constructor "ChangeTracker.QueryTrackingBehavior = QueryTrackingBehavior.NoTracking;"